clientlibs/js/src/functions/getExperimentCondition.ts

```diff
@@ -52,25 +76,51 @@ export class Assignment {
       return null;
     }
   }
+
+  public markDecisionPoint(status: MARKED_DECISION_POINT_STATUS, uniquifier?: string, clientError?: string) {
```

Comment:
Is this new public method on Assignment necessary? It looks like is just proxying the `Client.markExperimentPoint`, but I don't know what the benefit is to pass all of this `markObject` data this way?

Reply:
"we have removed the client.markExperimentPoint call" This is interesting, but unexpected! maybe we missed this in discussions @mswartzCL has set up some time to on Tues at 9 to meet, might be easier to chat in person to review changes in the lib API to know what the Java client library should be doing.

Reply:
I'm only noticing this in passing at a very high level, but: relying on "business logic" in the client libraries is, I think, a code smell.

We ought to strive for very simple client libraries, so as to encourage use of UpGrade from clients written in languages for which we don't provide a first-party library.

If you have to get a bunch of fiddly logic right in a client lib for an esoteric (or just unsupported) language, that's a deterrent to adoption.

Again, that may not be what's going on here at all, but sounded like it might be ... :)

Reply:
@danoswaltCL We have decided in the meeting that we will use the assignment object to mark the experiment and that is the reason we have removed the mark function directly from the upgrade client.
```
const client = new UpgradeClient();
client.markExperimentPoint(site, target, ....)           // this method is remove from the v5

const assingment = client.getDecisionPoint(site, target);
assignment.markExperimentPoint(....).                  // this method is added in v5
```

Reply:
Client mark call is back. Reverted those changes and stack is also rotated from `client.markDecisionPoint()` call

Reply:
@danoswaltCL I would add a commit to rename every `markExperimentPoint` to `markDecoisionPoint`